src/main/java/com/arcbees/analytics/client/ClientAnalytics.java

```diff
@@ -207,6 +207,13 @@ private native void nativeInit() /*-{
             }, i[r].l = 1 * new Date();
             a = s.createElement(o), m = s.getElementsByTagName(o)[0];
             a.async = 1;
+            // If current protocol is "file" we are on mobile
+            // So we explicitly specify https protocol for the script's source link
+            if ($wnd.location.protocol === "file:") {
+            	// CHECKSTYLE_OFF
+                g = "https:" + g;
+                // CHECKSTYLE_ON
+            }
             a.src = g;
             m.parentNode.insertBefore(a, m)
         })($wnd, $doc, 'script', '//www.google-analytics.com/analytics.js',
```

Comment:
Yes I think just add the https://  we force ssl by default anyway so that
shouldn't cause any issues.

On Thu, Jun 18, 2015 at 6:50 AM, Oleksandr Kruk <notifications@github.com>
wrote:

> In src/main/java/com/arcbees/analytics/client/ClientAnalytics.java
> <https://github.com/ArcBees/universal-analytics/pull/37#discussion_r32698966>
> :
>
> > @@ -207,6 +207,13 @@ private native void nativeInit() /*-{
> >              }, i[r].l = 1 * new Date();
> >              a = s.createElement(o), m = s.getElementsByTagName(o)[0];
> >              a.async = 1;
> > +            // If current protocol is "file" we are on mobile
> > +            // So we explicitly specify https protocol for the script's source link
> > +            if ($wnd.location.protocol === "file:") {
> > +            	// CHECKSTYLE_OFF
> > +                g = "https:" + g;
> > +                // CHECKSTYLE_ON
> > +            }
> >              a.src = g;
> >              m.parentNode.insertBefore(a, m)
> >          })($wnd, $doc, 'script', '//www.google-analytics.com/analytics.js',
>
> The only reason is to keep the protocol http/https depending on user's
> current protocol. If that's not important to you then I'll update it with
> https always, the code would be much cleaner also.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ArcBees/universal-analytics/pull/37/files#r32698966>.
>